src/test/requestUser.test.js

```diff
@@ -0,0 +1,263 @@
+import {
+  collection,
+  addDoc,
+  getDocs,
+  query,
+  where,
+  updateDoc,
+  arrayUnion,
+  arrayRemove,
+  doc,
+} from "firebase/firestore";
+
+import {
+  addUserRequest,
+  acceptRequest,
+  userPendingList,
+  userAcceptedList,
+  checkUserRequest,
+  checkUserPending,
+} from "@/firebase/requestUser";
+
+const tableName = "user_request";
+
+jest.mock("firebase/firestore", () => ({
+  getFirestore: jest.fn(),
+  collection: jest.fn(),
+  addDoc: jest.fn(),
+  query: jest.fn(),
+  getDocs: jest.fn(),
+  where: jest.fn(),
+  updateDoc: jest.fn(),
+  arrayUnion: jest.fn(),
+  arrayRemove: jest.fn(),
+}));
+
+describe("User Request logic", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  [
+    {
+      user_id: "1231",
+      pending_list: ["Walter", "Joe"],
+      accepted_list: ["White"],
+    },
+    {
+      uid: "1234",
+      pending_list: ["Joe"],
+      accepted_list: ["White"],
+    },
+  ];
+
+  it("test 1: Add fromUserID to pending list of toUserID.", async () => {
+    const toUserID = "12345";
+    const fromUserID = "ABCDE";
+
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              user_id: "12345",
+              pending_list: ["Walter", "Joe"],
+              accepted_list: ["White"],
+            },
+          ],
+        });
+      },
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    await addUserRequest(toUserID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(collection).toHaveBeenCalledWith(tableName);
+  });
+
+  it("test 2: Add fromUserID to pending list of toUserID.", async () => {
+    const toUserID = "12345";
+    const fromUserID = "ABCDE";
+
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              user_id: "12345",
+              pending_list: ["Walter", "Joe"],
+              accepted_list: ["White"],
+            },
+          ],
+        });
+      },
+      empty: (fn) => true,
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    await addUserRequest(toUserID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(addDoc).toHaveBeenCalledWith(collection(tableName), {
+      user_id: toUserID,
+      accepted_list: [],
+      pending_list: [fromUserID],
+    });
+    expect(collection).toHaveBeenCalledWith(tableName);
+  });
+
+  it("test 1: Add user to accepted list both users.", async () => {
```

Comment:
this also has the wrong test name, it has the same name as the test below it but only checks if we've searched up one of the users. can the name be updated to what this is testing?